packages/terra-search-field/src/SearchField.jsx

```diff
@@ -86,6 +103,10 @@ class SearchField extends React.Component {
   handleTextChange(event) {
     this.setState({ searchText: event.target.value });
```

Comment:
The way you're doing this, the state can become out of sync with the actual text in a controlled component. 
When a value and onChange is given to the input, if the user types something the onChange is triggered, we update the state here and then call the consumer's onChange. Then the consumer has the opportunity to modify that text and send down whatever they want as the value (ex. the user may have typed in a special character that the consumer needs to restrict that from being entered in the search field). This will update the text in the input, but now our state reflects what the user tried to type and not what is actually displayed in the input, so now the wrong text well be passed through onSearch.

Reply:
I added updating the state on componentDidUpdate if the value prop is different from the state so they stay in sync